internal/elasticsearch/index/index.go

```diff
@@ -0,0 +1,424 @@
+If specified, this mapping can include: field names, field data types (https://www.elastic.co/guide/en/elasticsearch/reference/current/mapping-types.html), mapping parameters (https://www.elastic.co/guide/en/elasticsearch/reference/current/mapping-params.html).
+**NOTE:** changing datatypes in the existing _mappings_ will force index to be re-created.`,
+			Type:             schema.TypeString,
+			Optional:         true,
+			DiffSuppressFunc: utils.DiffJsonSuppress,
+			ValidateFunc:     validation.StringIsJSON,
+			Default:          "{}",
+		},
+		"settings": {
+			Description: `Configuration options for the index. See, https://www.elastic.co/guide/en/elasticsearch/reference/current/index-modules.html#index-modules-settings.
+**NOTE:** Static index settings (see: https://www.elastic.co/guide/en/elasticsearch/reference/current/index-modules.html#_static_index_settings) can be only set on the index creation and later cannot be removed or updated - _apply_ will return error`,
+			Type:     schema.TypeList,
+			MaxItems: 1,
+			Optional: true,
+			Elem: &schema.Resource{
+				Schema: map[string]*schema.Schema{
+					"setting": {
+						Description: "Defines the setting for the index.",
+						Type:        schema.TypeSet,
+						Required:    true,
+						MinItems:    1,
+						Elem: &schema.Resource{
+							Schema: map[string]*schema.Schema{
+								"name": {
+									Description: "The name of the setting to set and track.",
+									Type:        schema.TypeString,
+									Required:    true,
+								},
+								"value": {
+									Description: "The value of the setting to set and track.",
+									Type:        schema.TypeString,
+									Required:    true,
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+		"settings_raw": {
+			Description: "All raw settings fetched from the cluster.",
+			Type:        schema.TypeString,
+			Computed:    true,
+		},
+	}
+
+	utils.AddConnectionSchema(indexSchema)
+
+	return &schema.Resource{
+		Description: "Creates Elasticsearch indices. See: https://www.elastic.co/guide/en/elasticsearch/reference/current/indices-create-index.html",
+
+		CreateContext: resourceIndexCreate,
+		UpdateContext: resourceIndexUpdate,
+		ReadContext:   resourceIndexRead,
+		DeleteContext: resourceIndexDelete,
+
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+
+		CustomizeDiff: customdiff.ForceNewIfChange("mappings", func(ctx context.Context, old, new, meta interface{}) bool {
+			o := make(map[string]interface{})
+			if err := json.NewDecoder(strings.NewReader(old.(string))).Decode(&o); err != nil {
+				return true
+			}
+			n := make(map[string]interface{})
+			if err := json.NewDecoder(strings.NewReader(new.(string))).Decode(&n); err != nil {
+				return true
+			}
+			log.Printf("[TRACE] mappings custom diff old = %+v new = %+v", o, n)
+
+			var isForsable func(map[string]interface{}, map[string]interface{}) bool
```

Comment:
Proper spelling here would be `isForceable`. English is a silly language.

Reply:
done in 73f3a7a